srml/support/src/metadata.rs

```diff
@@ -328,24 +351,36 @@ mod tests {
 						])
 					}
 				))),
-			}
-		]),
-		outer_dispatch: OuterDispatchMetadata {
-			name: DecodeDifferent::Encode("Call"),
-			calls: DecodeDifferent::Encode(&[
-				OuterDispatchCall {
-					name: DecodeDifferent::Encode("EventModule"),
-					prefix: DecodeDifferent::Encode("event_module"),
-					index: 0,
-				},
-				OuterDispatchCall {
-					name: DecodeDifferent::Encode("EventModule2"),
-					prefix: DecodeDifferent::Encode("event_module2"),
-					index: 1,
-				}
-			])
-		}
-	};
+				// lost DecodeDifferent::Encode("Module"), aka module metadata name
+				call: DecodeDifferent::Encode(FnEncode(||
+					CallMetadata {
+						name: DecodeDifferent::Encode("Call"),
+						functions: DecodeDifferent::Encode(&[]),
+					}
+				)),
+				outer_dispatch: DecodeDifferent::Encode(
+					FnEncodeModule("event_module2", |_|
+						Some(OuterDispatchCall {
+							name: DecodeDifferent::Encode("EventModule2"),
+							prefix: DecodeDifferent::Encode("event_module2"),
+							index: 1,
+						})
+					)
+				),
+				event: DecodeDifferent::Encode(
+					FnEncodeModule("event_module2", |_|
```

Comment:
"event_module2" is replicated multiple times, this should only added once to the metadata.

Reply:
yes, that is related to the other issue (running a runtime switch), as a static parameter I do not think it takes more space (it is not displayed in the resulting metadata).